### PR TITLE
Ternary operator

### PR DESCRIPTION
**Pull request by ylow**
Implements a ternary operator. A little like np.where. Extremely useful for replacing values in an SArray.

```
Returns an SArray with the same values as g with values above 10
clipped to 10

>>> g = SArray([6,7,8,9,10,11,12,13])
>>> SArray.where(g > 10, 10, g)
dtype: int
Rows: 8
[6, 7, 8, 9, 10, 10, 10, 10]

Returns an SArray with the same values as g with values below 10
clipped to 10

>>> SArray.where(g > 10, g, 10)
dtype: int
Rows: 8
[10, 10, 10, 10, 10, 11, 12, 13]

Returns an SArray with the same values of g with all values == 1
replaced by None

>>> g = SArray([1,2,3,4,1,2,3,4])
>>> SArray.where(g == 1, None, g)
dtype: int
Rows: 8
[None, 2, 3, 4, None, 2, 3, 4]
```

Also fixes a bunch of oddities with SArray.from_const.
Previously,
```
gl.SArray.from_const([1], 10, list)
```
will produce an SArray of
[[1.0],[1.0],[1.0],[1.0],[1.0]...]
converting each of the integers to a float.

Now it does the right thing.

### CODE REVIEW

**Comment by ylow** [comment]
@hershaw the ternary operator we discussed.

**Comment by hershaw** [comment]
This is great! Simpler and more flexible than the numpy implementation.

File: oss_src/sframe_query_engine/operators/ternary_operator.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+  }
+
+  static std::shared_ptr<planner_node> make_planner_node(
+      std::shared_ptr<planner_node> condition,
+      std::shared_ptr<planner_node> istrue,
+      std::shared_ptr<planner_node> isfalse) {
+    
+    return planner_node::make_shared(planner_node_type::TERNARY_OPERATOR, 
+                                     {},
```
Comment (haijieg) [inline oss_src/sframe_query_engine/operators/ternary_operator.hpp]: what's the type of this {}? clang might have warning/error for std::map

Comment (ylow) [inline oss_src/sframe_query_engine/operators/ternary_operator.hpp]: didn't seem to on my mac. But I can make it explicit.

**Comment by ylow** [comment]
@haijieg updated.

**Comment by haijieg** [comment]
LGTM

**Pull request merged**